app/controllers/api/sessions_controller.rb

```diff
@@ -0,0 +1,19 @@
+module Api
+  class SessionsController < BaseController
+    def create
+      auth_provider = AuthFactory.build(params)
+      user = auth_provider.find_user
+      user = auth_provider.create_user! unless user
```

Comment:
I would prefer a more explicit if statement here instead of the unless modifier... But anyway OK